ml-stat.py

```diff
@@ -229,6 +241,25 @@ def authors(self, mapping):
         remove_bots(people)
         return people
 
+    def timespan(self):
+        begin = self.root_msg.date()
+        last = begin
+        # messages are sorted by time, incidentally
+        for msg in self.msgs:
+            d = msg.date()
+            if msg.is_pwbot_command() or msg.is_pwbot_accept():
+                    last = d
+                    break
+            # for prolonged threads also enable "cutting looong tail" heuristic
+            if d - begin > datetime.timedelta(days=5):
+                # reply after a long period of silence
+                if d - last > 1.4 * (last - begin):
+                    break
+            last = d
+
+        # round up to one day
+        return max(last - begin, datetime.timedelta(days=1))
+
```

Comment:
perhaps we should add 1 day instead of just rounding up, to somehow account for the break between iterations